main.go

```diff
@@ -138,7 +138,7 @@ func main() {
 				// before actually setting up the watcher
 				logrus.WithError(err).Error("Error setting up watcher")
 			}
-			t := tailer.NewPathWatcher(path, nil, handlerFactory, stateRecorder)
```

Comment:
Nitpick: Can you make this its own named function (like the one in `watcherForPod` to make the code a bit more readable?